pkgs/build-support/go/module.nix

```diff
@@ -171,16 +171,32 @@ in
 
     inherit (go) GOOS GOARCH;
 
-    GOFLAGS = GOFLAGS
-      ++ lib.warnIf (lib.any (lib.hasPrefix "-mod=") GOFLAGS) "use `proxyVendor` to control Go module/vendor behavior instead of setting `-mod=` in GOFLAGS"
-        (lib.optional (!finalAttrs.proxyVendor) "-mod=vendor")
-      ++ lib.warnIf (builtins.elem "-trimpath" GOFLAGS) "`-trimpath` is added by default to GOFLAGS by buildGoModule when allowGoReference isn't set to true"
-        (lib.optional (!allowGoReference) "-trimpath");
+  env = args.env or { } // {
+
+    GOFLAGS = lib.concatStringsSep " " finalAttrs.goFlags;
```

Comment:
I'm not sure if we should do this from nix.

Can't we perhaps have `goFlags` be used as a bash array and set `GOFLAGS` later in bash?

Reply:
After examining the implementation again, it turns out the `GOFLAGS` is only used during the `buildPhase` provided by `buildGoModule`, and the `configurePhase` is still adjusting the flags.

Let's form the GOFLAGS at the beginning of the `buildPhase` then.